sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/procedures/ProcedureParameter.java

```diff
@@ -92,7 +92,7 @@ class Builder {
     private String defaultValueExpression;
     private String comment;
 
-    private Builder(Mode mode, String name, DataType dataType) {
+    public Builder(Mode mode, String name, DataType dataType) {
```

Comment:
We have `static Builder in(...)` in this file, can we add `out` and `inout` as well?

Reply:
SG

Reply:
now this can remain private?